spec/lib/metasploit/framework/credential_collection_spec.rb

```diff
@@ -323,6 +339,253 @@
       end
     end
 
+    context "when given additional_publics" do
+      let(:username) { nil }
+      let(:password) { nil }
+      let(:additional_publics) { [ "test_public" ] }
+
+      context "when :user_as_pass is true" do
+        let(:user_as_pass) { true }
+
+        # REMOVE BEFORE COMMIT currently failing
+        specify  do
+          expect { |b| collection.each(&b) }.to yield_successive_args(
+            Metasploit::Framework::Credential.new(public: "test_public", private: "test_public"),
+          )
+        end
+
+
+        context "when using password spraying" do
+          let(:password_spray) { true }
+
+          # REMOVE BEFORE COMMIT currently failing
+          specify  do
+            expect { |b| collection.each(&b) }.to yield_successive_args(
+              Metasploit::Framework::Credential.new(public: "test_public", private: "test_public"),
+            )
+          end
+        end
+      end
+
+      context "when :nil_passwords is true" do
+        let(:nil_passwords) { true }
+
+        # REMOVE BEFORE COMMIT: this option is ignored currently for additional_publics
+        specify do
+          expect { |b| collection.each(&b) }.to yield_successive_args(
+            Metasploit::Framework::Credential.new(public: "test_public", private: nil),
+          )
+        end
+      end
+
+      context "when using password spraying" do
+        let(:password_spray) { true }
+
+        context "when :blank_passwords and :nil_password are true" do
+          let(:blank_passwords) { true }
+          let(:nil_passwords) { true }
+
+          context "with 2 additional_publics" do
+            let(:additional_publics) { [ "test_public1", "test_public2" ] }
+
+            # REMOVE BEFORE COMMIT: fails because no pwd spraying
+            specify  do
+              expect { |b| collection.each(&b) }.to yield_successive_args(
+                Metasploit::Framework::Credential.new(public: "test_public1", private: ""),
+                Metasploit::Framework::Credential.new(public: "test_public2", private: ""),
+                Metasploit::Framework::Credential.new(public: "test_public1", private: nil),
+                Metasploit::Framework::Credential.new(public: "test_public2", private: nil),
```

Comment:
This fails because the password spraying option does not work with additional_publics.
It generates the correct list of credentials, but not in the correct order.